### PR TITLE
add get_messages  as a helper function.

### PR DESCRIPTION
**Pull request by tebadi**

### CODE REVIEW

**Comment by alexgleith** [comment]
I think we should move this to the aws tools, not just as a utils function in the dc_tools folder. Perhaps we create an `queues.py` file in there?

**Comment by tebadi** [comment]
Moved queue helper functions to odc.aws.

**Comment by alexgleith** [comment]
Hey, can you please remove the _cli app runner thing from this PR @tebadi

File: libs/aws/odc/aws/queues.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            WaitTimeSeconds=10,
+            MessageAttributeNames=["All"],
+        )
+
+        if len(messages) == 0 or (limit and count >= limit):
+            break
+        else:
+            for message in messages:
+                count += 1
+                yield message
```
Comment (alexgleith) [inline libs/aws/odc/aws/queues.py]: I think this should have a newline

File: libs/aws/odc/aws/queues.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    sqs = boto3.resource('sqs')
+    queue = sqs.get_queue_by_name(QueueName=queue_name)
+    return queue
+
+def publish_message(queue, msg):
+    resp = queue.send_message(
+        QueueUrl=queue.url,
+        MessageBody=msg
+    )
+    assert(resp['ResponseMetadata']['HTTPStatusCode'] == 200)
```
Comment (alexgleith) [inline libs/aws/odc/aws/queues.py]: Careful with parenthesis here. And this should probably be describing what happened and not just failing.
```suggestion
    assert resp['ResponseMetadata']['HTTPStatusCode'] == 200, "Failed to publish the message"
```

File: libs/aws/odc/aws/queues.py (hunk 1 above)
Comment (tebadi) [reply libs/aws/odc/aws/queues.py]: Added a newline.

File: libs/aws/odc/aws/queues.py (hunk 2 above)
Comment (tebadi) [reply libs/aws/odc/aws/queues.py]: Done.

**Comment by tebadi** [comment]
cli_app_runner changes should be excluded now.

**Review by alexgleith — APPROVED** [review]

File: apps/dc_tools/setup.py [hunk 3]
```diff
@@ -25,6 +25,8 @@
         'datacube',
         'odc_index',
         'odc_io',
+        'odc_aws'
+        'odc_aio'
```
Comment (Kirill888) [inline apps/dc_tools/setup.py:29]: is `odc_aio` legitimately needed? It pulls in `aiobotocore` which can be a pain.

File: libs/aws/odc/aws/queues.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        QueueUrl=queue.url,
+        MessageBody=msg
+    )
+
+    assert resp['ResponseMetadata']['HTTPStatusCode'] == 200, "Failed to publish the message"
+
+def get_messages(queue, limit):
+    count = 0
+    while True:
+        messages = queue.receive_messages(
```
Comment (Kirill888) [inline libs/aws/odc/aws/queues.py]: What happens when there is a Timeout, i.e. queue is empty? Should we catch that and translate into end of iteration.

File: libs/aws/odc/aws/queues.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            VisibilityTimeout=60,
```
Comment (Kirill888) [inline libs/aws/odc/aws/queues.py]: What is `VisibilityTimeout=`. Is it not something we want to expose as a configurable parameter?
Same goes for other timeout.

File: libs/aws/odc/aws/queues.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import boto3
+
+def get_queue(queue_name):
```
Comment (Kirill888) [inline libs/aws/odc/aws/queues.py]: code that creates s3 client supports configuring profile, region code, and accepting credentials, would be nice if this also had these features, there is code that creates session that takes care of all of that, one can then create sqs resource from that session.

Comment (alexgleith) [reply libs/aws/odc/aws/queues.py]: I think we can wait until we need it to deal with this.

File: libs/aws/odc/aws/queues.py (hunk 4 above)
Comment (alexgleith) [reply libs/aws/odc/aws/queues.py]: The timeout is how long a message takes to reappear on the queue.

**Pull request merged**